examples/steal_split/sources/steal_split.move

```diff
@@ -0,0 +1,1141 @@
+    const EPlayerHasDecisionSubmitted: u64 = 8;
+
+    struct ResouceAccountAddress has key {
+        addr: address
+    }
+
+    struct State has key {
+        next_game_id: u128,
+        games: SimpleMap<u128, Game>,
+        cap: SignerCapability,
+    }
+
+    struct WGBCOIN {}
+
+    struct Game has store, copy, drop {
+        prize_pool_amount: u64,
+        player_one: PlayerData,
+        player_two: PlayerData,
+        expiration_timestamp_in_seconds: u64,
+    }
+
+    struct PlayerData has store, copy, drop {
+        player_address: address,
+        decision_hash: Option<vector<u8>>,
+        salt_hash: Option<vector<u8>>,
+        decision: u64
+    }
+
+    struct CreateGameEvent has store, drop {
+        game_id: u128,
+        prize_pool_amount: u64,
+        player_one_address: address,
+        player_two_address: address,
+        expiration_timestamp_in_seconds: u64,
+        event_creation_timestamp_in_seconds: u64
+    }
+
+
+    struct SubmitDecisionEvent has store, drop {
+        game_id: u128,
+        player_address: address,
+        decision_hash: vector<u8>,
+        salt_hash: vector<u8>,
+        event_creation_timestamp_in_seconds: u64
+    }
+
+    struct RevealDecisionEvent has store, drop {
+        game_id: u128,
+        player_address: address,
+        decision: u64,
+        event_creation_timestamp_in_seconds: u64
+    }
+
+    struct ConcludeGameEvent has store, drop {
+        game_id: u128,
+        player_one_decision: u64,
+        player_two_decision: u64,
+        prize_pool_amount: u64,
+        event_creation_timestamp_in_seconds: u64
+    }
+
+    struct ReleaseFundsAfterExpirationEvent has store, drop {
+        game_id: u128,
+        player_one_decision: u64,
+        player_two_decision: u64,
+        prize_pool_amount: u64,
+        event_creation_timestamp_in_seconds: u64
+    }
+
+    fun init(account: &signer, ctx: &mut StorageContext) {
+        // let source_addr = signer::address_of(account);
+        let (signer, cap) = account::create_resource_account(ctx, account);
+        let resource_address = signer::address_of(&signer);
+
+        account_storage::global_move_to(ctx, account, ResouceAccountAddress {
+            addr: resource_address
+        });
+
+        let (burn_cap, freeze_cap, mint_cap) = coin::initialize<WGBCOIN>(account);
+
+        let coin = coin::mint(1000 * 1000 * 1000, &mint_cap);
+        coin::register<WGBCOIN>(account, ctx);
+        coin::register<WGBCOIN>(&signer, ctx);
+        coin::merge(coin::global_borrow_mut<WGBCOIN>(ctx, signer::address_of(account)), coin);
+
+        coin::destroy_burn_cap(burn_cap);
+        coin::destroy_freeze_cap(freeze_cap);
+        coin::destroy_mint_cap(mint_cap);
+
+        account_storage::global_move_to(ctx, &signer, State {
+            next_game_id: 0,
+            games: simple_map::create(),
+            cap
+        });
+
+        event::ensure_event_handle<CreateGameEvent>(ctx);
```

Comment:
Developers do not need to explicitly call `ensure_event_handle`, this method will be called implicitly by the contract when calling `event::emit`

Reply:
Thanks, I have fixed it.